### ISSUE TITLE
Operator & helm chart do not set `spec.template.spec.securityContext.{runAsUser, runAsGroup}` for pod-creating resources

### ISSUE THREAD

**Issue by aesca1er** [rook/rook#10546]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
rook operator creates pod-creating resources without securityContext runAsUser or runAsGroup
rook operator helm chart does not set securityContext runAsUser or runAsGroup
cephcluster CRD does not have a field for securityContext configuration for pod-creating resources
**Expected behavior:**
rook operator creates pod-creating resources with desired securityContext values
rook operator helm chart creates pod-creating resources with desired securityContext values
cephcluster CRD has a field securityContext configuration for pod-creating resources
**How to reproduce it (minimal and precise):**

deploy the rook operator and a cephcluster (optionally via helm charts as I did) on a v1.23+ kubeadm cluster

**File(s) to submit**:
https://github.com/rook/rook/blob/ba67a90f104826b18803b924f3b4c8c8da7db5ec/pkg/operator/ceph/csi/spec.go#L705
https://github.com/rook/rook/blob/ba67a90f104826b18803b924f3b4c8c8da7db5ec/pkg/operator/k8sutil/cmdreporter/cmdreporter.go#L91

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): n/a
* Kernel (e.g. `uname -a`): n/a
* Cloud provider or hardware configuration: n/a
* Rook version (use `rook version` inside of a Rook Pod): 1.9.6
* Storage backend version (e.g. for ceph do `ceph -v`): n/a
* Kubernetes version (use `kubectl version`): v1.23.5
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/Troubleshooting/ceph-toolbox/#interactive-toolbox)): n/a

**Comment by sp98** [comment]
@aesca1er What do you mean by `pod-creating resources`? Can you please elaborate this a bit more.

**Comment by aesca1er** [comment]
Pod creating resources are things like jobs or deployments.

**Comment by aesca1er** [comment]
@sp98 
> Do you mean Operator is not adding securityOontext to the Pods that it is creating?
Yes, and there is no option to configure them in the CephCluster CRD or operator helm chart.

**Comment by sp98** [comment]
Also wanted to check if starting privileged pods is something that might be helpful in the mean time. See [here](https://github.com/rook/rook/blob/1ae867049b49079b76696e68ee9b8f30216528bd/deploy/examples/operator.yaml#L522)

**Comment by aesca1er** [comment]
> @aesca1er Also wanted to check if starting privileged pods is something that might be helpful in the mean time. See [here](https://github.com/rook/rook/blob/1ae867049b49079b76696e68ee9b8f30216528bd/deploy/examples/operator.yaml#L522)

It looks like this only applies to the ceph mon and osd pods. The first issue I have encountered is regarding the securityContext for the csi-discover job.
